src/esp/gfx/Renderer.cpp

```diff
@@ -48,11 +49,7 @@ struct Renderer::Impl {
             scene::SceneGraph& sceneGraph,
             RenderCamera::Flags flags) {
     ASSERT(visualSensor.isVisualSensor());
```

Comment:
In the argument list, `visualSensor` is a reference of `VisualSensor` instance. 
That means `isVisualSensor` always returns true. 
I would remove this ASSERT.